server/src/main/java/com/cloud/server/StatsCollector.java

```diff
@@ -411,7 +461,42 @@ private void init(Map<String, String> configs) {
 
     }
 
-    class HostCollector extends ManagedContextRunnable {
+    /**
+     * Configures the database name according to the URI path. For instance, if the URI is as influxdb://address:port/dbname, the database name will be 'dbname'.
+     */
+    protected String configureDatabaseName(URI uri) {
+        String dbname = StringUtils.removeStart(uri.getPath(), "/");
+        if (StringUtils.isBlank(dbname)) {
+            return DEFAULT_DATABASE_NAME;
+        } else {
+            return dbname;
+        }
+    }
+
+    /**
+     * Configures the port to be used when connecting with the stats collector service.
+     * Default values are 8086 for influx DB and 2003 for GraphiteDB.
+     * Throws URISyntaxException in case of non configured port and external StatsType
+     */
+    protected int configureExternalStatsPort(URI uri) throws URISyntaxException {
```

Comment:
Are you configuring a port? Or, are you retrieving a port? I mean, when you configure, I would expect a set something somewhere. Here, I see you checking a few things, returning some data according to some conditions, and using some defaults.

Reply:
Good catch, the name might mislead. I will update to `retrieveExternalStatsPortFromUri`